src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvThread.cs

```diff
@@ -302,7 +300,7 @@ private void OnPost()
                 wasWork = DoPostWork();
                 wasWork = DoPostCloseHandle() || wasWork;
                 loopsRemaining--;
-            } while (wasWork && loopsRemaining > 0);
+            } while (!_stopping && wasWork && loopsRemaining > 0);
```

Comment:
What are you trying to achieve here? If you are afraid of this running during the second uv_run, you would want this as a while instead of a do/while condition.